libraries/rib-base/src/main/java/com/badoo/ribs/clienthelper/childawareness/ChildAwareRegistry.kt

```diff
@@ -0,0 +1,69 @@
+package com.badoo.ribs.clienthelper.childawareness
+
+import androidx.lifecycle.Lifecycle
+import com.badoo.ribs.core.Rib
+import com.badoo.ribs.core.plugin.NodeAware
+import com.badoo.ribs.core.plugin.SubtreeChangeAware
+import kotlin.reflect.KClass
+
+/**
+ * This interface provides simplified access to the Nodes children
+ * when you need to create connection between them without any intermediate processing.
+ */
+interface ChildAwareRegistry : SubtreeChangeAware, NodeAware {
+
+    fun <T : Rib> whenChildBuilt(
+        lifecycle: Lifecycle,
+        child: KClass<T>,
+        callback: ChildCallback<T>,
+    )
+
+    fun <T : Rib> whenChildAttached(
+        lifecycle: Lifecycle,
+        child: KClass<T>,
+        callback: ChildCallback<T>,
+    )
+
+    fun <T1 : Rib, T2 : Rib> whenChildrenBuilt(
+        lifecycle: Lifecycle,
+        child1: KClass<T1>,
+        child2: KClass<T2>,
+        callback: ChildrenCallback<T1, T2>,
+    )
+
+    fun <T1 : Rib, T2 : Rib> whenChildrenAttached(
+        lifecycle: Lifecycle,
+        child1: KClass<T1>,
+        child2: KClass<T2>,
+        callback: ChildrenCallback<T1, T2>,
+    )
```

Comment:
These won't account for the cases when one should be only built, but the other only attached. But adding all combinations would be an overkill and confusing too. I think we need to collapse these two methods into one (only attached), and cache outputs in the build->attached period to properly solve this.

Reply:
> cache outputs in the build->attached period to properly solve this
It is absolutely different and complicated task that requires writing own implementation of UnicastSubject(Relay). Also it will make built callback meaningless.

It is even possible to have 2 children in built, but not in attached state? As I know, when you create new child, it is built and attached in one run.

Reply:
One can be already attached, the other only built, when you want to set up listening to outputs to not lose any. But you can have it the other way around too.

I find that the two methods (onChildBuilt and onChildAttached) are a workaround in the first place, and not something we would need otherwise. Also they are easy to miss / mix up. So maybe it's worth to solve it for both cases, and win a leaner API. See #293.